airbyte-integrations/connectors/source-snowflake/unit_tests/test_schema_builder.py

```diff
@@ -1,31 +1,267 @@
+    assert convert_utc_to_time_zone(utc_date, offset_hours) == expected
+
+    # Test case: fractional negative offset
+    utc_date = datetime(2023, 1, 1, 12, 0, 0)
+    offset_hours = -3.75
+    expected = '2023-01-01T08:15:00.000000-03:45'
+    assert convert_utc_to_time_zone(utc_date, offset_hours) == expected
+
+    # Test case: large positive offset
+    utc_date = datetime(2023, 1, 1, 12, 0, 0)
+    offset_hours = 14
+    expected = '2023-01-02T02:00:00.000000+14:00'
+    assert convert_utc_to_time_zone(utc_date, offset_hours) == expected
+
+    # Test case: large negative offset
+    utc_date = datetime(2023, 1, 1, 12, 0, 0)
+    offset_hours = -12
+    expected = '2023-01-01T00:00:00.000000-12:00'
+    assert convert_utc_to_time_zone(utc_date, offset_hours) == expected
+
+
+def test_convert_utc_to_time_zone_date():
+    # Test case: positive offset
+    utc_date = datetime(2023, 1, 1, 12, 0, 0)
+    offset_hours = 3
+    expected = datetime.strptime('2023-01-01T15:00:00.000000+03:00', '%Y-%m-%dT%H:%M:%S.%f%z')
+    assert convert_utc_to_time_zone_date(utc_date, offset_hours) == expected
+
+    # Test case: negative offset
+    utc_date = datetime(2023, 1, 1, 12, 0, 0)
+    offset_hours = -5
+    expected = datetime.strptime('2023-01-01T07:00:00.000000-05:00', '%Y-%m-%dT%H:%M:%S.%f%z')
+    assert convert_utc_to_time_zone_date(utc_date, offset_hours) == expected
+
+    # Test case: zero offset
+    utc_date = datetime(2023, 1, 1, 12, 0, 0)
+    offset_hours = 0
+    expected = datetime.strptime('2023-01-01T12:00:00.000000+00:00', '%Y-%m-%dT%H:%M:%S.%f%z')
+    assert convert_utc_to_time_zone_date(utc_date, offset_hours) == expected
+
+    # Test case: fractional positive offset
+    utc_date = datetime(2023, 1, 1, 12, 0, 0)
+    offset_hours = 5.5
+    expected = datetime.strptime('2023-01-01T17:30:00.000000+05:30', '%Y-%m-%dT%H:%M:%S.%f%z')
+    assert convert_utc_to_time_zone_date(utc_date, offset_hours) == expected
+
+    # Test case: fractional negative offset
+    utc_date = datetime(2023, 1, 1, 12, 0, 0)
+    offset_hours = -3.75
+    expected = datetime.strptime('2023-01-01T08:15:00.000000-03:45', '%Y-%m-%dT%H:%M:%S.%f%z')
+    assert convert_utc_to_time_zone_date(utc_date, offset_hours) == expected
+
+    # Test case: large positive offset
+    utc_date = datetime(2023, 1, 1, 12, 0, 0)
+    offset_hours = 14
+    expected = datetime.strptime('2023-01-02T02:00:00.000000+14:00', '%Y-%m-%dT%H:%M:%S.%f%z')
+    assert convert_utc_to_time_zone_date(utc_date, offset_hours) == expected
+
+    # Test case: large negative offset
+    utc_date = datetime(2023, 1, 1, 12, 0, 0)
+    offset_hours = -12
+    expected = datetime.strptime('2023-01-01T00:00:00.000000-12:00', '%Y-%m-%dT%H:%M:%S.%f%z')
+    assert convert_utc_to_time_zone_date(utc_date, offset_hours) == expected
+
+
+class TestFormatField(unittest.TestCase):
+
+    def test_none_field_type(self):
+        self.assertEqual("value", format_field("value", None))
```

Comment:
nit: no need to change that but for the newly added tests, we can rely on the assert statement as this is printed way better when running with the pytest runner than `self.assertEqual`

Reply:
Replaced in these tests